src/ParquetMsg.chpl

```diff
@@ -267,40 +267,16 @@ module ParquetMsg {
     var fileErrorCount:int = 0;
     var fileErrorMsg:string = "";
     var sizes: [filedom] int;
-    var ty = getArrType(filenames[filedom.low],
-                        dsetlist[dsetdom.low]);
+    var types: [dsetdom] ArrowTypes;
     var rnames: list((string, string, string)); // tuple (dsetName, item type, id)
 
-    for dsetname in dsetnames do {
+    for (dsetidx, dsetname) in zip(dsetdom, dsetnames) do {
         for (i, fname) in zip(filedom, filenames) {
             var hadError = false;
             try {
-                // not using the type for now since it is only implemented for ints
-                // also, since Parquet files have a `numRows` that isn't specifc
-                // to dsetname like for HDF5, we only need to get this once per
-                // file, regardless of how many datasets we are reading
-                sizes[i] = getArrSize(fname);
-            } catch e: FileNotFoundError {
-                fileErrorMsg = "File %s not found".format(fname);
-                pqLogger.error(getModuleName(),getRoutineName(),getLineNumber(),fileErrorMsg);
-                hadError = true;
-                if !allowErrors { return new MsgTuple(fileErrorMsg, MsgType.ERROR); }
-            } catch e: PermissionError {
-                fileErrorMsg = "Permission error %s opening %s".format(e.message(),fname);
-                pqLogger.error(getModuleName(),getRoutineName(),getLineNumber(),fileErrorMsg);
-                hadError = true;
-                if !allowErrors { return new MsgTuple(fileErrorMsg, MsgType.ERROR); }
-            } catch e: DatasetNotFoundError {
-                fileErrorMsg = "Dataset %s not found in file %s".format(dsetname,fname);
-                pqLogger.error(getModuleName(),getRoutineName(),getLineNumber(),fileErrorMsg);
-                hadError = true;
-                if !allowErrors { return new MsgTuple(fileErrorMsg, MsgType.ERROR); }
-            } catch e: SegArrayError {
-                fileErrorMsg = "SegmentedArray error: %s".format(e.message());
-                pqLogger.error(getModuleName(),getRoutineName(),getLineNumber(),fileErrorMsg);
-                hadError = true;
-                if !allowErrors { return new MsgTuple(fileErrorMsg, MsgType.ERROR); }
+                (sizes[i], types[dsetidx]) = (getArrSize(fname), getArrType(fname, dsetname));
             } catch e : Error {
```

Comment:
I'm not sure that I would say it is more robust error handling. I'm not sure there is much of a value add for throwing specific classes of errors when the content of the error (which is what the user actually sees) remains the same. In my opinion, defining a bunch of new Parquet error classes like was done for HDF5 (see `src/ServerErrors.chpl`) is more work for little value, but if this does add value in a way that I'm not aware of I think it would be worth discussing.

Reply:
I tend to agree that simplifying it down would be fine provided there is no added details provided from various error types. In some instances though, I do think that granular level detail can be helpful when debugging and this could be helpful.

The message is added after stating "Other error in accessing...". The `e.message()` should add the details, but it may be harder to identify during a debug situation. 

Not sure if we could restructure the current message in a way to that would summarize more clearly since we are using a catch all for it. Maybe something like `'Parquet Access Error: %s, %s'.format(fname, e.message())`